### PR TITLE
fix(streaming): fail closed on malformed Gemini SSE parts

### PR DESCRIPTION
**Pull request by jeong-sik**
## Summary

- Make Gemini SSE parsing a closed result: malformed JSON and malformed candidate/Part shapes become explicit wire parse failures with the raw payload preserved.
- Validate the supported Gemini Part boundary before event conversion so unsupported or malformed function calls, media, and payload combinations cannot disappear silently.
- Convert both parser failures and event-decoder failures to OAS SSEParseFailed events in complete_stream.
- Update streaming callers and regression coverage; remove the old synthetic test expectation that treated a Gemini oneof violation as valid.

## Verification

- ocamlc -stop-after parsing: all six touched OCaml files
- ocamlformat --check: all six touched OCaml files
- git diff --check
- focused Dune build: lib/llm_provider/llm_provider.cma
- focused test executables:
  - test/test_streaming_edge_cases.exe: 15 passed
  - test/test_backend_gemini.exe: 55 passed
  - test/test_gemini_edge_cases.exe: 5 edge cases passed

## Contract evidence

Gemini GenerateContent responses expose candidate content and finish reasons, and function calls are structured Parts. Unsupported wire shapes must remain observable as a transport parse failure:
- https://ai.google.dev/api/generate-content
- https://ai.google.dev/gemini-api/docs/generate-content/function-calling

This PR is OAS-only and does not introduce any MASC dependency.

### CODE REVIEW

File: lib/llm_provider/streaming.ml [hunk 1]
```diff
@@ -1843,47 +1843,269 @@ type gemini_chunk =
...
+    | value ->
+      Error (Printf.sprintf "%s:not_object(got %s)" path (Json_util.json_type_name value))
+  in
+  let part = `Assoc fields in
+  let* () =
+    let supported_payloads = [ "text"; "functionCall"; "inlineData" ] in
+    let metadata_fields = [ "thought"; "thoughtSignature" ] in
+    let unsupported_field =
+      List.find_opt
+        (fun (key, _) -> not (List.mem key (supported_payloads @ metadata_fields)))
```
Comment (jeong-sik) [inline lib/llm_provider/streaming.ml]: P1 — this whitelist conflates a valid Gemini Part with malformed SSE.

The current official Generate Content contract also returns `executableCode` / `codeExecutionResult` for Code Execution and `toolCall` / `toolResponse` when built-in tools are circulated; `functionResponse` and `fileData` are valid Part variants in the same API. OAS currently declares Gemini `supports_code_execution = true` and document/audio/video capabilities, but this parser rejects those fields as `unsupported_field`, and the event projector has no typed representation for them. Therefore a valid provider response is surfaced as `SSEParseFailed` and later `Malformed_payload`, which loses the distinction the PR is intended to preserve.

This needs one of two explicit root fixes before this PR can be adversarial-pass: implement the full typed request/sync/stream/replay path for the declared Gemini surfaces, or narrow the declared capability and add a distinct typed unsupported-feature result. Do not broaden the whitelist without an exhaustive event projection; that would silently drop the valid Part after parsing. Official references: https://ai.google.dev/api/generate-content and https://ai.google.dev/gemini-api/docs/generate-content/function-calling

Comment (jeong-sik) [reply lib/llm_provider/streaming.ml]: SHA `f2dc53cf3d70c72ea12dcffdc7d43858e3584e34`에서 P1을 근본 경계로 해소했어용.

- Gemini의 `supports_code_execution` 선언은 false로 좁혔어용.
- 응답에서 실제 반환되는 `executableCode`, `codeExecutionResult`, `toolCall`, `toolResponse`는 shape 검증 후 `SSEUnsupportedPart → Stream_unsupported_part → Capability_mismatch`로 보존해용.
- malformed shape/unknown Part는 계속 parse failure이고, malformed sibling이 있으면 unsupported 판정보다 우선해용.
- `functionResponse`는 사용자 입력, `fileData`도 입력용이며 생성 파일 응답은 `inlineData`라 response parser의 공식 unsupported 집합에는 넣지 않았어용.
- 이벤트/누산기/HTTP 오류/telemetry 전 경로와 회귀 테스트를 추가했고 포맷·파싱·diff 검사를 통과했어용.

공식 근거: https://ai.google.dev/gemini-api/docs/generate-content/tool-combination · https://ai.google.dev/gemini-api/docs/generate-content/code-execution

새 exact-head CI `30812087244` 확인 후 스레드를 닫을게용.

Comment (jeong-sik) [reply lib/llm_provider/streaming.ml]: exact HEAD f2dc53cf3d70c72ea12dcffdc7d43858e3584e34 기준으로 확인했습니다.

공식 Gemini 문서의 response Part는 executableCode/codeExecutionResult 및 built-in toolCall/toolResponse입니다. functionResponse는 다음 user turn의 input이고 fileData도 request input이므로 response unsupported 집합에서 제외했습니다.

현재 parser는 네 response part를 shape 검증 후 Gemini_unsupported_part -> SSEUnsupportedPart -> Stream_unsupported_part -> Capability_mismatch로 kind/raw를 보존합니다. 내부 shape 오류와 unknown Part는 Gemini_parse_failed/Malformed_payload로 남고, 같은 envelope 안 malformed sibling이 있으면 unsupported보다 우선합니다.

supports_code_execution=false는 serializer/request projection/stream/replay의 완전한 typed 경로가 아직 없기 때문이며, 전체 지원은 #2931에서 별도 추적합니다. CI exact run 30812087244 required checks green입니다.
